[iOS][CI] Add a new job to support custom build

xta0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32323 [iOS][CI] Add a new job to support custom build**

### Summary

Since we have released the custom build in 1.4.0, it's time to setup a CI for that. This PR adds a new iOS job to the iOS builds. To save time, It only runs the arm64 build. 

### Test Plan

- Don't break any iOS jobs
- Custom Build works.

Differential Revision: [D19451342](https://our.internmc.facebook.com/intern/diff/D19451342)

xta0 committed:
[iOS][CI] Add a new job to support custom build

xta0 committed:
Update on "[iOS][CI] Add a new job to support custom build"

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit d0947504:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=d09475040bef3873085ee97b95fa4381635d2a1c).

### 1 failure *not* recognized by patterns:
| Job | Step | Status |
| --- | --- | --- |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>pytorch_python_doc_push</sup> | <sup>Doc Build and Push</sup> | New in PR |


---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

xta0 committed:
Update on "[iOS][CI] Add a new job to support custom build"


### Summary

This PR adds a new iOS job to the iOS builds. To save time, It only runs the arm64 build. 

### Test Plan

- Don't break any iOS jobs
- Custom Build works.

xta0 committed:
Update on "[iOS][CI] Add a new job to support custom build"


### Summary

This PR adds a new iOS job to the iOS builds. To save time, It only runs the arm64 build. 

### Test Plan

- Don't break any iOS jobs
- Custom Build works.
